### ISSUE TITLE
New enhancements for POS

### ISSUE THREAD

**Issue by bkrajendra** [jekkos/opensourcepos#566]
Few suggestion for next milestone:
1) Raise Purchase Order
2) Inward and Make goods receipt note (GRN)

I'll add more to these. I would myself like to work on these.

**Comment by daN4cat** [comment]
Purchasing side enhancement sounds a good idea, although expands a bit the basic POS story but it's good.

Do you plan to have a full PO cycle management?
What I mean is:
* raise PO
* get PO confirmation from supplier (my wife's case for instance can see differences between the order and the confirmation because they might have decided not to produce something due to too low overall quantity, or because they have supplying issues)
* receive delivery (add delivery note in pdf format)
* store invoice (store in pdf format) with payment terms and deadline
* close the business with payment(s)

Once this is done probably some integration with the accounting http://webzash.org/ can happen. Just a thought.

**Comment by daN4cat** [comment]
BTW, I marked that 3.1. I don't think we should aim at adding that to 3.0 as 3.0 was just a major infrastructure and UI overhaul.

**Comment by arteiro** [comment]
I liked the idea of using http://webzash.org/, but could have something simpler as posted in #574  to record Expenses from rent, electric power, phone, et al. Being able to generate a report with revenue and expenses in a single system. In my craft store I'm using OSPOS and Another with cash flow, so I have to record the sale in OSPOS and CASH FLOW, and es expenses only in CASH FLOW.
I am using this: demonstration http://paulocollares.com.br/livro_caixa_demo/.

**Comment by daN4cat** [comment]
@bkrajendra just wondering if you progressed this topic and if had in mind to make any contribution to the project

**Comment by alvin101** [comment]
I looked at webzash, it is simple and basic. I think as you pursue a back office that you should look at a system that already offers PO management among other features.  A connector or data sync to a system like http://frontaccounting.com/ would take care of a lot of the features ask for by users here.

**Comment by daN4cat** [comment]
frontaccounting could be functional but is ugly. I don't know how they can keep up with those type of user experiences nowadays. That's why I made a huge push to move OSPOS to the next UI level because I couldn't stand to the uglyness of the past anymore.

With regards to PO, it's a matter of taste. Some could like webzash as pure accounting and OSPOS for stock / sales / po management.
So far I couldn't find a valid PO solution that simply allows you to create a purchase item, attach an invoice scan, set a reminder for the payments and get the story closed when the payment is done.

**Comment by alvin101** [comment]
@daN4cat You can change the look and feel of frontaccount with one of the many themes or a new theme, IU is only skin deep. I just look at the functions, things like drill down financial reports, and foreign currency transactions. However I do not see any way to attache documents to PO's something I also need.  https://github.com/phreebooks/PhreeBooksERP has a way to attache a file to a PO, but only one, I need multiple attachments. However PhreeBooks dev has slowed, Dave the main dev no longer supports it, Rena is the only dev left and I am sure he is busy making a living.  I do like the way http://www.dolibarr.org/ handles PO's but I do not like the accounting side of it, you can buy modules to do accounting but I am not ready to jump in. Both these last two systems mentioned come with a POS.

At the end of the day I think a back office is the way to go for these features that we all need.  Hans over at POSper was working on a back office https://sourceforge.net/p/posper/discussion/606131/thread/b4856118/ but I never saw a working copy. 

Something else you might look at https://www.symmetricds.org/ I tried it a few years ago but I just was not tech savoy enough to make it work.

**Comment by daN4cat** [comment]
Sure many good solutions, but my problem is that they try to solve too many problems that we don't have.
Anyway, this is going beyond the original post. I would like to understand if the original poster made any progress.

**Comment by bkrajendra** [comment]
@daN4cat, @alvin101,  I do understand the need for Purchase order Management and other feature request I posted is really important. But as I'm currently bzy with my other embedded (IoT) projects, I'm struggling for time. Additionally I am very much familiar with **Laravel** framework, and learning new one is real pain :( !
Still I'm working on these aspects and will soon come up with solution.

**Comment by daN4cat** [comment]
@SteveIreland have you ever read this discussion thread?

**Comment by SteveIreland** [comment]
Yes. I've read it a few times but not in the last month.  Thanks for bringing it to my attention again.  I'll make sure that we've got it covered in the requirements..

**Comment by daN4cat** [comment]
@SteveIreland I moved this to 3.3.0. It's better to close a couple of small things and then release 3.2.0 then hang it to something bigger that will eventually come.

**Comment by SteveIreland** [comment]
Thanks @daN4cat.  That's what I was expecting.

**Comment by SteveIreland** [comment]
I'm okay with moving this to 4 because I'm going to have an nightmare on my hands rebasing the completed work over the 3.3 release so that I can continue development, but what is your current target date for the release of 3.3?

**Comment by jekkos** [comment]
Well I was going through the list of issues that have 3.3 as due date, seeing that most of them don't have any owner or development done. I'd say that once we have attributes and GST we could do a 3.3? Unless there's other work coming in by the time those two are finished.

**Comment by jekkos** [comment]
One interesting solution for accounting is odoo. It's opensource seems to be more like a full ERP system but opensource. Haven't had the chance to try it yet but it looks very complete.

**Comment by jekkos** [comment]
Also it has a set of connectors by which you can [import data](http://odoo-connector.com/) from other systems (like google sheets, csv files or the like)

**Comment by odiea** [comment]
Webzash has not been updated for a year and Odoo looks like it costs money to use.

**Comment by SteveIreland** [comment]
It's developed in Python.

**Comment by SteveIreland** [comment]
Reviewed a YouTube introductory video on Odoo.

Interesting product.  Did you bring it to our attention because you are thinking of heading that direction?

I've been looking for an excuse to expend the effort to do a deep dive into Python.  If you're thinking of heading that direction I'll join you on your journey.

**Comment by jekkos** [comment]
Well in fact I'm looking for a solution to book invoices related to the orders we enter in ospos now. But as discussed previously I don't really know if it's a good idea to try to integrate all that in ospos. Also I know eg @daN4cat is just using google sheets for this and others use other solutions, so was looking for something that would fit here nicely.
I think i'd be okay keeping ospos for regular sales and stock while in the meantime syncing with a system like this, especially because it's oss an it has been there for quite some time (openerp was the name before the rebrand)

**Comment by SteveIreland** [comment]
I'm going to go ahead and take a look at what they are making available in the community edition.  Some of the features that are discussed in the area of inventory, purchasing, and vendor invoice management would take me a year to plug in to OSPOS by myself with the time allowed ... maybe more.  It would also be nice to have a final solution for accounts receivable.

**Comment by SteveIreland** [comment]
@jekkos I just learned that Odoo is locked in to PostgreSQL with no other alternatives.  I checked their primary competitor which seems to be ERPNext and they are pretty much locked into MariaDB which is more compatible with OSPOS.

Do you have any thoughts or opinions regarding Odoo vs ERPNext and PostgreSQL vs MariaDB?

For some reason major ISP hosts don't seem to be fans of PostgreSQL so I have  a little concern there.

**Comment by bernd-wechner** [comment]
> PostgreSQL

I come in only on a hit searching the repo of postgresql support as I have the opposite problem, or the same if you will. That is, all our existing services and infrastructure and Postgresql bound and I find significant FOSS tools often don't support it alas, this one included, though there is a hint here that it might:

https://github.com/opensourcepos/opensourcepos/blob/6d8890f61ee7bb51150d34239576b0bf28275394/application/config/database.php#L22

Can't comment on major ISP hosts (as we self host) but postgresql is a pretty key player:

https://db-engines.com/en/ranking_trend/system/MySQL%3BPostgreSQL

and it seems to me that supporting MysQL and Postgresql is the 2021 equivalent of coding support for Internet Explorer/Firefox and Chrome 20 years ago ;-). They are not perfectly SQL compatible (close but not perfect) and the market is strongly divided. And hence I do see mentions of a dbdriver as noted above, which is a clue that OSPOS is built from the ground up to be database agnostic but only a clue, not evidence of any gravity (as no quick search of a repo and reading on a few lines of source in a random file ever can be)

**Comment by daN4cat** [comment]
There is no PostGres support but happy for you to contribute and maintain it here on ospos.

**Comment by bernd-wechner** [comment]
Big maybe there ... ;-). Would play with the docker build first to get a feel for it. And Tyro support is more important really for us,
